components/main/SearchBar.tsx

```diff
@@ -1,37 +1,61 @@
-import { useState } from 'react';
-import { SearchData } from '../../types/mainType';
+import { useState, useCallback } from 'react';
 import { getData } from '../../utils/axios';
 
 export default function SearchBar() {
   const [keyword, setKeyword] = useState<string>('');
-  const [showSearchBox, setShowSearchBox] = useState<boolean>(false);
-  const [searchResult, setSearchResult] = useState<SearchData | null>(null);
+  const [showDropDown, setShowDropDown] = useState<boolean>(false);
+  const [searchResult, setSearchResult] = useState<string[] | null>(null);
 
-  const showSearchBoxHandler = () => {
-    setShowSearchBox(!showSearchBox);
+  const showDropDownHandler = () => {
+    setShowDropDown(!showDropDown);
   };
 
   const keywordHandler = (event: React.ChangeEvent<HTMLInputElement>) => {
     setKeyword(event.target.value);
   };
 
+  const getSearchResultHandler = async () => {
+    const filteredKeyword = keyword.replace(/[^a-zA-Z0-9\-]/g, '');
+    if (filteredKeyword) {
+      try {
+        const data = await getData(
+          `/pingpong/users/searches/${filteredKeyword}`
+        );
+        setSearchResult(data.users);
+      } catch (e) {}
+    }
+  };
+
   return (
     <div>
       <input
         type='text'
+        value={keyword}
         onChange={keywordHandler}
-        onFocus={showSearchBoxHandler}
-        onBlur={showSearchBoxHandler}
+        onKeyUp={makeDebounce(getSearchResultHandler, 800)}
+        onFocus={showDropDownHandler}
+        onBlur={showDropDownHandler}
         placeholder='챔피온 검색하기'
       />
       <button>&#128269;</button>
-      {showSearchBox && (
+      {showDropDown && (
         <div>
-          {searchResult
-            ? searchResult.users.map((item: string) => <div>{item}</div>)
-            : '검색 결과가 없습니다.'}
+          {!searchResult || !keyword
+            ? '검색 결과가 없습니다.'
+            : searchResult?.map((item, i) => <div key={i}>{item}</div>)}
```

Comment:
```suggestion
          {searchResult && keyword
            ? searchResult?.map((item, i) => <div key={i}>{item}</div>)}
            : '검색 결과가 없습니다.'```

이렇게 하면 가독성을 더 높일 수 있을것 같습니다 :)